roles/openshift_adm/tasks/shutdown.yml

```diff
@@ -0,0 +1,64 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# Initiates a graceful shutdown of the OpenShift cluster.
+
+
+- name: Check for regenerated certificate file.
+  when:
+    - not cifmw_openshift_adm_dry_run
+  ansible.builtin.stat:
+    path: "{{ cifmw_openshift_adm_cert_expire_date_file }}"
+  register: _regenerated_file
+
+# Current implementation works only once hence the check.
+- name: Regenerate the API certificates.
+  when:
+    - not _regenerated_file.stat.exists | default(false)
+    - not cifmw_openshift_adm_dry_run
+  ansible.builtin.include_tasks: api_cert.yml
+
+- name: Gather the list of nodes
+  kubernetes.core.k8s_info:
+    kind: Node
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    validate_certs: false
+  register: _nodes
+
+- name: Mark the nodes as unschedulable.
+  when:
+    - not cifmw_openshift_adm_dry_run
+  kubernetes.core.k8s_drain:
+    name: "{{ item }}"
+    state: cordon
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    validate_certs: false
+  loop: "{{ _nodes.resources | map(attribute='metadata.name') | list }}"
+
+- name: Initiate shutdown
+  when:
+    - not cifmw_openshift_adm_dry_run
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.shell:
+    cmd: >-
+      for node in $(oc get nodes -o jsonpath='{.items[*].metadata.name}');
+      do oc debug node/${node} -- chroot /host shutdown -h 3; done
+
+- name: Wait till shutdown is completed on all the nodes
+  ansible.builtin.pause:
+    minutes: 5
```

Comment:
can't we get a better way to detect that? Maybe as a follow-up though. (missed it in my previous reading, sorry)

Reply:
Looking at alternatives